src/core/font_substitutions.js

```diff
@@ -456,19 +456,29 @@ function generateFont(
  * @param {String} baseFontName The font name to be substituted.
  * @param {String|undefined} standardFontName The standard font name to use
  *   if the base font is not available.
+ * @param {String} type The font type.
  * @returns an Object with the CSS, the loaded name, the src and the style.
  */
 function getFontSubstitution(
   systemFontCache,
   idFactory,
   localFontPath,
   baseFontName,
-  standardFontName
+  standardFontName,
+  type
 ) {
   if (baseFontName.startsWith("InvalidPDFjsFont_")) {
     return null;
   }
 
+  if (
+    (type === "TrueType" || type === "Type1") &&
+    /^[A-Z]{6}\+/.test(baseFontName)
```

Comment:
I just noticed that we fortunately have unit tests for this function at https://github.com/mozilla/pdf.js/blob/master/test/unit/font_substitutions_spec.js so aside from the reference test it might be good to, if possible of course, add a unit test for this too since that could surface any regressions sooner since the unit tests are much faster to run.